osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -186,6 +187,37 @@ public void OnReleased(KeyBindingReleaseEvent<T> e)
 
         #endregion
 
+        #region Keys per second Counter Attachment
+
+        public void Attach(KeysPerSecondCalculator calculator)
+        {
+            if (calculator == null) return;
+
+            var listener = new ActionListener();
+
+            KeyBindingContainer.Add(listener);
+
+            calculator.Listener = listener;
```

Comment:
Am I the only one that finds this convoluted? There's this `ActionListener`, which is being added to the `KeyBindingContainer`, with its own `OnNewInput` event, and it needs to be set on this `calculator` to work properly. Wouldn't it be simpler to pass the calculator to the listener, and directly invoke methods on it whenever a new input comes in, rather than have this one extra layer of event binding?

Reply:
Fixed in 9dc8065